gcore/src/error.rs

```diff
@@ -0,0 +1,56 @@
+// This file is part of Gear.
+//
+// Copyright (C) 2022 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+pub use gear_core_errors::{ExtError, MemoryError, MessageError, TerminationReason};
+
+#[cfg(feature = "codec")]
+mod sys {
+    extern "C" {
+        pub fn gr_error(data: *mut u8);
+    }
+}
+
+#[must_use]
+#[repr(transparent)]
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct SyscallError {
+    len: u32,
+}
+
+impl SyscallError {
+    pub fn into_result(self) -> Result<(), ExtError> {
+        if self.len == 0 {
+            Ok(())
+        } else {
+            #[cfg(feature = "codec")]
```

Comment:
I guess `cfg!` will increase readability. What do you think? Also if we bear in mind costs of not removing code (`cfg!` doesn't remove unused branches)

Reply:
> Also if we bear in mind costs of not removing code

I think compiler will remove it (dead code elimination)

Reply:
I decided to leave it as is because compiler always check the code in branch with `cfg!()` and we have conditional imports

Reply:
if there are no cons from time cost perspective, then using `cfg!()` will give you more readability, right?

Reply:
alright